linux/shim/init.go

```diff
@@ -60,15 +62,27 @@ func newInitProcess(context context.Context, path, namespace string, r *shimapi.
 		}
 		options = *v.(*runcopts.CreateOptions)
 	}
+
+	rootfs := filepath.Join(path, "rootfs")
+	// count the number of successful mounts so we can undo
+	// what was actually done rather than what should have been
+	// done.
+	nrRootMounts := 0
```

Comment:
Actually, the more i look at this why don't you just loop until you receive a mount error that the path is not mounted instead of keeping this count around?